src/main/java/org/isf/menu/service/MenuIoOperations.java

```diff
@@ -175,44 +185,48 @@ public boolean isGroupNamePresent(String groupName) throws OHServiceException {
 
 	/**
 	 * Inserts a new {@link User} in the DB
-	 * 
 	 * @param user - the {@link User} to insert
 	 * @return the new {@link User} added to the DB
 	 * @throws OHServiceException When failed to create user
 	 */
 	public User newUser(User user) throws OHServiceException {
+		user.setDeleted(false);
 		return repository.save(user);
 	}
 
 	/**
 	 * Updates an existing {@link User} in the DB
-	 * 
 	 * @param user - the {@link User} to update
-	 * @return new {@link User}
+	 * @return the updated {@link User}
 	 * @throws OHServiceException When failed to update user
 	 */
-	public boolean updateUser(User user) throws OHServiceException {
-		return repository.updateUser(user.getDesc(), user.getUserGroupName(), user.getUserName()) > 0;
+	public User updateUser(User user) throws OHServiceException {
+		repository.updateUser(user.getDesc(), user.getUserGroupName(), user.isDeleted(), user.getUserName());
+
+		return getUserByName(user.getUserName());
 	}
 
 	/**
 	 * Updates the password of an existing {@link User} in the DB
-	 * 
 	 * @param user - the {@link User} to update
-	 * @return {@code true} if the user has been updated, {@code false} otherwise.
+	 * @return the {@link User} that has been deleted
 	 * @throws OHServiceException When failed to update the password
 	 */
-	public boolean updatePassword(User user) throws OHServiceException {
-		return repository.updatePassword(user.getPasswd(), user.getUserName()) > 0;
+	public User updatePassword(User user) throws OHServiceException {
+		ensureUserNotDeleted(user.getUserName());
+		repository.updatePassword(user.getPasswd(), user.getUserName());
+		return getUserByName(user.getUserName());
 	}
 
 	/**
 	 * Deletes an existing {@link User}
-	 * 
 	 * @param user - the {@link User} to delete
 	 * @throws OHServiceException When failed to delete user
 	 */
 	public void deleteUser(User user) throws OHServiceException {
+		if (!user.isDeleted()) {
+			ensureUserNotDeleted(user.getUserName());
+		}
 		repository.delete(user);
 	}
```

Comment:
why the if condition here? should not be enough `ensureUserNotDeleted(user.getUserName());` like in the other methods?

Reply:
Your've just said in your previous comments:

> I don't think we should disable this button, because one may need to finally hard-delete the group

> I don't think we should disable buttonEdit and buttonDelete buttons, because one may need to restore the group or hard-delete it (once empty), hence we need to update the GroupEdit window with the new checkbox